### ISSUE TITLE
Still being maintained?

### ISSUE THREAD

**Issue by ConstantinoSchillebeeckx** [andialbrecht/sqlparse#624]
Is this project still being maintained? There are 172 open issues, 18 PRs and the last commit was in Dec 2020 :(

**Comment by andialbrecht** [comment]
yes, just busy at work

Am Mi., 26. Mai 2021 um 14:35 Uhr schrieb Constantino Schillebeeckx <
***@***.***>:

**Comment by ConstantinoSchillebeeckx** [comment]
Anything we can do to help out?

**Comment by tobymao** [comment]
I've created a full sql parser / transpiler from scratch and am actively developing it. It has a full tokenizer / parser and doens't use regex.

It can transpile from various sql dialects to other sql dialects as well as things like formatting and parsing things out.

https://github.com/tobymao/sqlglot

**Comment by f-raffa** [comment]
> Anything we can do to help out?

Yeah, actually if there was something we could do to help out I'd like to contribute...
Because of my job I worked I lot on the code and I would also have some changes to propose for discussion.
Would you be interested?

**Comment by f-raffa** [comment]
> I've created a full sql parser / transpiler from scratch and am actively developing it. It has a full tokenizer / parser and doens't use regex.
> 
> It can transpile from various sql dialects to other sql dialects as well as things like formatting and parsing things out.
> 
> https://github.com/tobymao/sqlglot

@tobymao: I checked your project and I was wondering if you were planning to integrate any formatting capabilities to it?

**Comment by tobymao** [comment]
@f-raffa i have a pretty mode, and you can chose the indent and padding levels

**Comment by f-raffa** [comment]
@tobymao: I see. I'd like simply to parse and format a sql statement without transpiling it.
Does your app provide such a functionality? May you please share the command that I should use?

**Comment by tobymao** [comment]
sqlglot.transpile("select * from x", read='presto', pretty=True)

**Comment by AhlamHani** [comment]
@tobymao this helped me to fix a bug!!

```python
    sql = sqlparse.format(sql, strip_whitespace=True, strip_comments=True).strip()
    sql = sqlglot.transpile(sql, read=Dialects.MYSQL, write=Dialects.MYSQL, pretty=True)[0]
```

the reason that `keyword_case='upper'` in sqlparse has a bug, it does not distinguish between table names and SQL commands, so your table name can be transformed to uppercase just because you are using a reserved name as a table name

here is my SQL query:

```sql

SELECT
    t.name AS tenant_name
FROM
    tenant t
        JOIN
    salik_account sa ON t.id = sa.tenant_id
        JOIN
    transaction tr ON sa.id = tr.salik_account_id
WHERE
    tr.status = 'completed';
```

using `sqlparse.format(sql, strip_whitespace=True, strip_comments=True, keyword_case='upper').strip()` also converted `transaction` table to `TRANSACTION`
